samples/drivers/dac/src/main.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2020 Libre Solar Technologies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/printk.h>
+#include <drivers/dac.h>
+
+#define DAC_DEVICE_NAME		DT_ALIAS_DAC1_LABEL
+#define DAC_RESOLUTION		12
+#define DAC_1ST_CHANNEL_ID	1
+
+static const struct dac_channel_cfg dac_ch1_cfg = {
+	.channel_id  = DAC_1ST_CHANNEL_ID,
+	.buffered = false,
+};
+
+void main(void)
+{
+	struct device *dac_dev = device_get_binding(DAC_DEVICE_NAME);
+
+	if (!dac_dev) {
+		printk("Cannot get DAC device\n");
+		return;
+	}
+
+	int ret = dac_channel_setup(dac_dev, &dac_ch1_cfg);
+
+	if (ret != 0) {
+		printk("Setting up of channel 1 failed with code %d\n", ret);
+		return;
+	}
+
+	printk("Generating sawtooth signal at DAC channel 1.\n");
+	while (1) {
+		/* 12-bit DAC, valid digital input from 0 to 4095 */
+		for (int i = 0; i < 4096; i++) {
+			dac_write_value(dac_dev, DAC_1ST_CHANNEL_ID, i, 12);
```

Comment:
`12` should be `DAC_RESOLUTION`. The interval for the for-loop should match the resolution (here it is hardcoded for 12 bit resolution).